qurro/qarcoal.py

```diff
@@ -31,6 +31,12 @@ def filter_and_join_taxonomy(feat_table, taxonomy, num_string, denom_string):
         denom_df: pd.DataFrame of denominator features x samples
     """
 
+    feat_table = feat_table.copy()
+    feat_table = feat_table.astype(pd.SparseDtype("float", fill_value=0))
+
+    # drop samples with total 0 features
+    feat_table = feat_table.loc[:, (feat_table != 0).any(axis=0)]
```

Comment:
Regarding removing 0-count samples -- are these samples still present in the output artifact, or are they excluded entirely? It would be good to have documentation explaining this. (I thought we had defined behavior for this kind of thing but I'm blanking on this right now.)

...actually wait wouldn't these samples already get filtered out by the `# drop columns (samples) in which no feature(s) matching string is present` stuff? That should be before the to-dense conversion, right?

Reply:
So the line to which you are referring occurs *after* the `NaNs` are introduced, so for whatever reason I think this has to be moved upstream if we go along with this.